File: CS/CustomQueryInReportDesigner/CustomQueryWizardClasses.cs
using System.Collections.Generic;
using DevExpress.XtraReports.Web;
using DevExpress.DataAccess.Sql;

namespace CustomQueryInReportDesigner {
    public class QueryInfo {
        public string Name { get; set; }
        public string Sql { get; set; }

        public CustomSqlQuery CreateCustomSqlQuery() {
            return new CustomSqlQuery() {
                Name = this.Name,
                Sql = this.Sql
            };
        }
    }

    public class DataSourceInfo {
        public string Name { get; set; }
        public string ConnectionName { get; set; }
        public List<QueryInfo> Queries { get; set; }

        public SqlDataSource CreateSqlDataSource() {
            SqlDataSource dataSource = new SqlDataSource() {
                Name = this.Name,
                ConnectionName = this.ConnectionName
            };

            if (Queries != null) {
                foreach (QueryInfo query in Queries) {
                    dataSource.Queries.Add(query.CreateCustomSqlQuery());
                }
            }

            dataSource.RebuildResultSchema();
            return dataSource;
        }
    }

    public class CustomQueryWizardModel {
        public List<DataSourceInfo> DataSources { get; set; }

        public void CreateDataSources(ASPxReportDesigner reportDesigner) {
            reportDesigner.DataSources.Clear();
            if (DataSources != null) {
                foreach(DataSourceInfo dataSource in DataSources) {
                    reportDesigner.DataSources.Add(dataSource.Name, dataSource.CreateSqlDataSource());
                }
            }
        }
    }
}

File: VB/CustomQueryInReportDesigner/CustomQueryWizardClasses.vb
Imports System.Collections.Generic
Imports DevExpress.XtraReports.Web
Imports DevExpress.DataAccess.Sql

Namespace CustomQueryInReportDesigner
	Public Class QueryInfo
		Public Property Name() As String
		Public Property Sql() As String

		Public Function CreateCustomSqlQuery() As CustomSqlQuery
			Return New CustomSqlQuery() With {.Name = Me.Name, .Sql = Me.Sql}
		End Function
	End Class

	Public Class DataSourceInfo
		Public Property Name() As String
		Public Property ConnectionName() As String
		Public Property Queries() As List(Of QueryInfo)

		Public Function CreateSqlDataSource() As SqlDataSource
			Dim dataSource As New SqlDataSource() With {.Name = Me.Name, .ConnectionName = Me.ConnectionName}

			If Queries IsNot Nothing Then
				For Each query As QueryInfo In Queries
					dataSource.Queries.Add(query.CreateCustomSqlQuery())
				Next query
			End If

			dataSource.RebuildResultSchema()
			Return dataSource
		End Function
	End Class

	Public Class CustomQueryWizardModel
		Public Property DataSources() As List(Of DataSourceInfo)

		Public Sub CreateDataSources(ByVal reportDesigner As ASPxReportDesigner)
			reportDesigner.DataSources.Clear()
			If DataSources IsNot Nothing Then
				For Each dataSource As DataSourceInfo In DataSources
					reportDesigner.DataSources.Add(dataSource.Name, dataSource.CreateSqlDataSource())
				Next dataSource
			End If
		End Sub
	End Class
End Namespace